src/open_producten/producttypen/tests/test_load_upn.py

```diff
@@ -0,0 +1,101 @@
+import os
+from io import StringIO
+from unittest.mock import patch
+
+from django.core.management import CommandError, call_command
+from django.test import SimpleTestCase, TestCase
+
+from open_producten.producttypen.management.parsers import CsvParser
+from open_producten.producttypen.models import UniformeProductNaam
+
+
+class TestLoadUPNCommand(TestCase):
+
+    def call_command(self, *args, **kwargs):
+        out = StringIO()
+        call_command(
+            "load_upl",
+            *args,
+            stdout=out,
+            stderr=StringIO(),
+            **kwargs,
+        )
+        return out.getvalue()
+
+    def test_load_csv_with_correct_columns(self):
+        file_name = "data/upl.csv"
+        path = os.path.join(os.path.dirname(__file__), file_name)
+        result = self.call_command(path)
+        self.assertEqual(result, f"Importing upn from {path}...\nDone (1 objects).\n")
+        self.assertEqual(UniformeProductNaam.objects.count(), 1)
+        self.assertEqual(
+            UniformeProductNaam.objects.first().naam, "aangifte vertrek buitenland"
+        )
+        self.assertEqual(
+            UniformeProductNaam.objects.first().uri,
+            "http://standaarden.overheid.nl/owms/terms/AangifteVertrekBuitenland",
+        )
+
+    def test_load_csv_with_incorrect_columns(self):
+        file_name = "data/wrong-upl.csv"
+        path = os.path.join(os.path.dirname(__file__), file_name)
+
+        with self.assertRaisesMessage(CommandError, "'URI' does not exist in csv"):
+            self.call_command(path)
+
+
+class TestCSVParser(SimpleTestCase):
+
+    def setUp(self):
+        self.parser = CsvParser()
+
+    def test_parser_returns_error_when_format_is_not_csv(self):
+        with self.assertRaisesMessage(Exception, "File format is not csv"):
+            self.parser.parse("abc.txt")
+
+    @patch("open_producten.producttypen.management.parsers.NamedTemporaryFile")
+    @patch("open_producten.producttypen.management.parsers.CsvParser.process_csv")
+    def test_parser_with_url(self, mock_process_csv, mock_NamedTemporaryFile):
```

Comment:
```suggestion
    import requests_mock
    requests_mock.Mocker()
    def test_parser_with_url(self, mocker):
        with open("data/upl.csv") as f:
            mocker.get(
                "https://www.abc.com/abc.csv",
                text=f.read(),
            )
```

I don't think this test and the one after are particularly useful TBH. 
In this one anyway, you should mock the endpoint like above (not guaranteed to be working).

You can also use the [VCRpy](https://vcrpy.readthedocs.io/en/latest/) and record and *replay* the response to a real endpoint, since this is a publicly available thing? (keep in mind that these will probably have a lot of UPNs, buy you could just validate the first one and the count)

Example in Objects API:
* [implmentation](https://github.com/maykinmedia/objects-api/blob/master/src/objects/token/tests/test_admin.py)
* [files created](https://github.com/maykinmedia/objects-api/tree/master/src/objects/token/tests/cassettes)